### PR TITLE
add test-results timeline

### PR DESCRIPTION
**Pull request by nmalyschkin**
TODO:
- [x] move inline styles to CSS

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "clean-webpack-plugin": "^3.0.0",
     "concurrently": "^5.1.0",
     "css-loader": "^3.6.0",
+    "csstype": "^3.0.7",
```
Comment (FlorianRappl) [inline package.json]: Why do we need this?

File: src/widgets/test-results/timeline/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import * as React from "react";
+import { OBSERVATION_INTERPRETATION } from "../loadPatientTestData/helpers";
+import { ObsRecord } from "../loadPatientTestData/types";
+import styles from "./timeline.scss";
+import * as CSS from "csstype";
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/helpers.tsx]: Why do we need this? React already brings typings for that.

File: src/widgets/test-results/timeline/helpers.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  children = undefined,
+  style
+}) => (
+  <TimeSlotsInner style={style}>
+    <div>{children}</div>
+  </TimeSlotsInner>
+);
+
+export const GridItems = React.memo<{
+  sortedTimes: string[];
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/helpers.tsx]: Can we consistently use `Array<T>`? Thanks!

File: src/widgets/test-results/timeline/useScroll.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    } else {
+      toBeExecuted = true;
+      lastArgs = args;
+    }
+  };
+
+  return throttledFunc;
+};
+
+const useScrollIndicator = (xThreshold, yThreshold) => {
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/useScroll.ts]: Types?

File: src/widgets/test-results/timeline/useScroll.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+
+const makeThrottled = (func, time = 1000) => {
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/useScroll.ts]: What type is `func`?

File: src/widgets/test-results/timeline/useScroll.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  let waiting = false;
+  let toBeExecuted = false;
+  let lastArgs = null;
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/useScroll.ts]: Type?

File: src/widgets/test-results/timeline/helpers.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  obs: ObsRecord[];
+}>(({ sortedTimes, obs }) => (
+  <>
+    {sortedTimes.map((_, i) => {
+      if (!obs[i]) return <TimelineCell text={"--"} />;
+      const interpretation = obs[i].meta.assessValue(obs[i].value);
+      return (
+        <TimelineCell text={obs[i].value} interpretation={interpretation} />
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/helpers.tsx]: Missing `key`

File: src/widgets/test-results/timeline/helpers.tsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  </TimeSlotsInner>
+);
+
+export const GridItems = React.memo<{
+  sortedTimes: string[];
+  obs: ObsRecord[];
+}>(({ sortedTimes, obs }) => (
+  <>
+    {sortedTimes.map((_, i) => {
+      if (!obs[i]) return <TimelineCell text={"--"} />;
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/helpers.tsx]: Missing `key`

File: src/widgets/test-results/timeline/timeline.component.tsx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    dataColumns={timeColumns.length}
+    padding
+    style={{ gridColumn: "span 2" }}
+  >
+    {Object.entries(rowData).map(([title, obs]) => {
+      const {
+        meta: { unit = "", range = "" }
+      } = obs.find(x => !!x);
+      return (
+        <>
```
Comment (FlorianRappl) [inline src/widgets/test-results/timeline/timeline.component.tsx]: Missing `key`

**Review by FlorianRappl — COMMENTED** [review]
Goes in the right direction - well done! :beers:

File: src/widgets/test-results/timeline/helpers.tsx (hunk 3 above)
Comment (nmalyschkin) [reply src/widgets/test-results/timeline/helpers.tsx]: I added an eslint rule for that. Would it make sense to add it to the other repos as well?

File: .eslintrc [hunk 10]
```diff
@@ -3,5 +3,9 @@
       "ts-react-important-stuff",
       "plugin:prettier/recommended",
       "plugin:react-hooks/recommended"
-    ]
+    ],
+    "plugins": ["@typescript-eslint"],
+    "rules": {
+      "@typescript-eslint/array-type": ["error", { "default": "generic" }]
```
Comment (FlorianRappl) [inline .eslintrc]: Love it!

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks! :beers:

**Pull request merged**